components/core/src/clp/CurlGlobalInstance.cpp

```diff
@@ -0,0 +1,43 @@
+#include "CurlGlobalInstance.hpp"
+
+#include <mutex>
+
+#include <curl/curl.h>
+
+#include "CurlOperationFailed.hpp"
+#include "ErrorCode.hpp"
+
+namespace clp {
+CurlGlobalInstance::CurlGlobalInstance() {
+    std::lock_guard<std::mutex> const global_lock{m_global_mutex};
+    if (0 == m_num_living_instances) {
+        if (auto const err{curl_global_init(CURL_GLOBAL_ALL)}; CURLE_OK != err) {
```

Comment:
```suggestion
        if (auto const err{curl_global_init(CURL_GLOBAL_ALL)}; 0 != err) {
```

Although `CURLE_OK` is defined as 0, the [docs](https://curl.se/libcurl/c/curl_global_init.html) say to test for non-zero, so there might be some backward incompatibility with `CURLE_OK`.